addons/account/models/account_move_line.py

```diff
@@ -1526,6 +1526,14 @@ def write(self, vals):
         line_to_write = self
         vals = self._sanitize_vals(vals)
         for line in self:
+            # Disallow modifying readonly move line fields on a posted move
+            move_state = line.move_id.state
+            unmodifiable_fields = (
+                'invoice_date', 'date', 'journal_id', 'company_id', 'move_name', 'product_id', 'partner_id', 'tax_ids', 'date_maturity', 'product_uom_id',
+                'price_unit', 'discount', 'price_subtotal', 'price_total', 'quantity')
+            readonly_fields = [val for val in vals if val in unmodifiable_fields]
+            if not self._context.get('skip_readonly_check') and move_state == "posted" and readonly_fields:
+                raise UserError(_("You cannot modify the following readonly move line fields on a posted move: %s", ', '.join(readonly_fields)))
```

Comment:
This is not safe, especially for computed fields.
It should also see that no field that would trigger a recomputation is going to be written.

This can probably be done by using `_field_will_change` and by checking the triggers inside of that function; it should be done for the checks on the lock dates and hash as well actually...